numpy/core/src/multiarray/descriptor.c

```diff
@@ -1072,7 +1072,13 @@ _convert_from_dict(PyObject *obj, int align)
                 goto fail;
             }
             offset = PyInt_AsLong(off);
-            PyTuple_SET_ITEM(tup, 1, off);
+            if (PyErr_Occurred()) {
```

Comment:
I mean our own (or the python version we mostly call there) NpyPyIntAsIntp (or whate it was), which specificially does not allow floats.